### DISCUSSION TITLE
Select - sekundærtekst

### DISCUSSION THREAD

**Discussion by Murstam** [fremtind/jokul#2328] [Annet]
Vi har behov for en sekundærtekst i select (kunne vært dropdown), i vårt tilfelle er det org. nr som støtter bedriftsnavnet slik at det bli lettere å velge riktig bedrift hvis du har tilgang til flere.

Hvis noen flere har behov for det samme eller mener det brude være en ny komponent, si fra :)
<img width="1507" alt="Skjermbilde 2021-10-05 kl  15 38 32" src="https://user-images.githubusercontent.com/47025023/136034484-e4c0f9d3-b115-41f0-b0ba-140f9c6c38f3.png">

**Comment by wkillerud** [comment]
Vi er i en flytteprosess hvor vi prøver å holde temaer som er på diskusjonsnivå til Discussions-fanen i GitHub 😄 Hopper over til Discussions-taben med denne.

**Comment by wkillerud** [comment]
Jeg kan ikke helt huske å ha sett noen andre steder i Fremtind som har brukt dette patternet på inventory-runden til @onordbo og meg. Det sagt så er magefølelsen min at dette er noe som burde vært mulig med Jøkul som en variant av enten Select eller Autocomplete (kanskje mer det siste i dette tilfellet).

**Reply by piofinn** [reply]
> Autocomplete

Innspill her: "vanlig" Select kan også gjøres søkbar i Jøkul, så da er det kanskje fornuftig å ha der også?

**Reply by wkillerud** [reply]
Det gir mening. Det blir vel en slags variant av "noe du kan velge fra en meny", og både Select og Autocomplete har en sånn "meny".

**Comment by onordbo** [comment]
+1 for @wkillerud sin magefølelse her, vi jobber primært med løsninger for finansielle institutsjoner som sjelden identifiseres utelukkende basert på en navneverdi, og dette er ofte et naturlig mønster som går igjen for systemer med store/lange oversikter.

**Comment by karimarkhus** [comment]
Vi har en lignende greie i Flyt @Murstam  denne er utviklet men ikke i Jøkul

![Screenshot 2021-10-08 at 14 45 26](https://user-images.githubusercontent.com/91268511/136559337-1667e77d-5b1d-4be2-89ee-f9d1dd15cd5b.png)
.

**Reply by joms** [reply]
Den ser egentlig ut som den har 3 nivåer (bold, ikke-bold, undertekst). Er det riktig?

Off-topic må jeg si jeg liker veldig godt fokusmarkeringen sammenlignet med den som er i Jøkul. Mye tydligere hvilke element som har fokus. Noen som har tenkt det samme?

**Reply by onordbo** [reply]
Hvordan ser den som er i Jøkul ut, @joms? Er det denne pila som er hele markeringa? 🤔 

![image](https://user-images.githubusercontent.com/4263132/136750508-01f4721e-7195-4ba1-990d-7ee8ec65b1e7.png)

**Reply by wkillerud** [reply]
Stemmer @onordbo, kun pila: https://jokul.fremtind.no/komponenter/select

**Reply by onordbo** [reply]
Hmmm, det var ikke imponerende tydelig akkurat. Er det litt vel custom, relativt til det "vanlige" designmønsteret for fokusering?

**Reply by wkillerud** [reply]
Det er ikke det vanligste jeg ser rundtom nei. Native på macOS er at det fokuserte valget får "fokusfargen" som bakgrunn (i mitt tilfelle blått) med passende tekstfarge (hvitt hos meg). Tror det er ganske vanlig, alternativt en focus-ring.

**Reply by piofinn** [reply]
Jeg testet ut en versjon med bakgrunnsfarge for litt siden (og fet skrift for den valgte verdien). Select i Jøkul hadde original markering på den måten, men siden vi hadde færre farger å jobbe med ble det ikke superfint 😅 Så ble det piler under siste overhaling. Men nå som vi har en litt mer fleksibel fargepalett tenker jeg det er en god idé å utforske bakgrunnsfarge igjen!

**Reply by piofinn** [reply]
Kjapp test med Varde som farge, sammenliknet med native på Mac

https://user-images.githubusercontent.com/25739615/136939670-08553ffe-919b-49c6-b24e-9432082c6f69.mov

**Reply by joms** [reply]
Det ser mye bedre ut imo. Personlig mener jeg at hover-effekten på teksten er for dårlig også (jada, jeg vet den tilfredsstiller kontraster osv 😅 ) og at den ikke synes godt nok.

**Reply by onordbo** [reply]
Som @joms sier, vi er kanskje innafor på UU-krav, men det betyr ikke at vi ikke kan gjøre det sjukt mye bedre—noe dette definitivt er, @piofinn

**Comment by ChrisAtFremtind** [comment]
Ville bare skyte inn at vi i BSF teamet også har nytte for en lignende komponent, utvikling er ennå ikke starter på den.

<img width="1029" alt="Skjermbilde 2021-10-14 kl  10 28 23" src="https://user-images.githubusercontent.com/88424441/137281604-4dfec7f7-63ac-45ed-96f7-371918fc1d65.png">

**Comment by HavardPede** [comment]
+1 for denne saken. I SAM har vi select for kunder, der vi har fnr som sekundær tekst. Hadde vært veldig glad hvis vi får inn dette også i jøkul select. 

![image](https://user-images.githubusercontent.com/30959515/137319061-3603a398-25bc-4d6f-9bc7-6a397bfc221f.png)
